server/game/cards/02_SHD/upgrades/MandalorianArmor.ts

```diff
@@ -0,0 +1,27 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { UpgradeCard } from '../../../core/card/UpgradeCard';
+import { Trait } from '../../../core/Constants';
+import { Card } from '../../../core/card/Card';
+
+export default class MandalorianArmor extends UpgradeCard {
+    protected override getImplementationId() {
+        return {
+            id: '3514010297',
+            internalName: 'mandalorian-armor'
+        };
+    }
+
+    public override setupCardAbilities() {
+        this.setAttachCondition((card: Card) => !card.hasSomeTrait(Trait.Vehicle));
+
+        this.addWhenPlayedAbility({
+            title: 'Give a Shield token.',
```

Comment:
nit: this is ambiguous, it could mean "give a shield token to any unit". Needs to indicate that the shield token will go to the attached uit